widget/charts/src/components/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,381 @@
+                   * index passed to our tooltipData
+                   */
+                  const shouldHavePartialOpacity =
+                    !shouldBeHighlighted && tooltipOpen;
+
+                  const barColor =
+                    colorBlockchainMap.get(barStack.key) || bar.color;
+
+                  return (
+                    <rect
+                      key={`bar-stack-${barStack.index}-${bar.index}`}
+                      x={bar.x}
+                      y={bar.y}
+                      height={bar.height}
+                      width={bar.width}
+                      fill={barColor}
+                      opacity={shouldHavePartialOpacity ? 0.5 : 1}
+                      onClick={(event) => handleBarClick(event, bar, index)}
+                      onMouseLeave={handleMouseLeave}
+                      onMouseMove={(event) =>
+                        handleMouseMove(event, bar, index)
+                      }
+                    />
+                  );
+                })
+              );
+            }}
+          </BarStack>
+        </Group>
+
+        <AxisBottom
+          top={yMax + margin.top}
+          left={margin.left + 10}
+          scale={dateScale}
+          hideAxisLine
+          hideTicks
+          numTicks={numBottomAxis}
+          tickValues={bottomAxisValue}
+          tickFormat={(d) => getDayOfMonth(d)}
+          tickLabelProps={() => ({
+            fontSize: isMobile ? 10 : 12,
+            fill: '#727272',
+            textAnchor: 'middle',
+          })}
+        />
+
+        <AxisLeft
+          hideAxisLine
+          hideTicks
+          numTicks={isMobile ? 3 : 5}
+          top={margin.top}
+          left={margin.left}
+          scale={valueScale}
+          tickFormat={(d) => compactNumberFormat(Number(d))}
+          tickLabelProps={() => ({
+            fontSize: isMobile ? 10 : 12,
+            fill: '#727272',
+            textAnchor: 'middle',
+          })}
+        />
+      </svg>
+
+      {tooltipOpen && tooltipData && (
+        <TooltipInPortal
+          top={tooltipTop}
+          left={tooltipLeft}
+          style={{ backgroundColor: 'transparent', boxShadow: 'none' }}>
+          <TooltipContainer
+            ref={tooltipRef}
+            style={{ boxShadow: TOOLTIP_SHADOW }}>
+            {tooltipData.bar.bar.data.date && (
+              <TooltipInfoRow>
+                <div>
+                  {dayjs
+                    .utc(tooltipData.bar.bar.data.date)
+                    .local()
+                    .format('YYYY/MM/DD')
+                    .toString()}
+                </div>
+                <div>
+                  {AmountConverter(
+                    Number(getTotalValue(tooltipData.bar.bar.data).toFixed(2))
+                  )}
+                </div>
+              </TooltipInfoRow>
+            )}
+            {Array.from(colorBlockchainMap).map((mapItem) => {
```

Comment:
We should not use the name of `blockchain` anywhere in the code.
It is better to use a more general name.
In general, we can consider each blockchain as a bucket.

For example:
`colorBlockchainMap`  -> `colorBucketMap`
`blockchainItem`   ----->   `bucketItem` 
...